packages/modular-scripts/src/esbuild-scripts/plugins/workerFactoryPlugin.ts

```diff
@@ -0,0 +1,141 @@
+import * as esbuild from 'esbuild';
+import * as path from 'path';
+import getModularRoot from '../../utils/getModularRoot';
+
+// This plugin builds Web Workers on the fly and exports them to use like worker-loader for Webpack 4: https://v4.webpack.js.org/loaders/worker-loader/
+// The workers are not inlined, a new file is generated in the bundle. Only files *imported* with the *.worker pattern are matched.
+// The workers are trampolined to avoid CORS errors.
+// This will be deprecated in the future when esbuild supports the Worker signature: see https://github.com/evanw/esbuild/issues/312
+// And will probably end up being compatible with Webpack 5 support https://webpack.js.org/guides/web-workers
+
+function createPlugin(): esbuild.Plugin {
+  const plugin: esbuild.Plugin = {
+    name: 'worker-factory-plugin',
+    setup(build) {
+      // This stores built workers for later use
+      const workerBuildCache: Map<string, esbuild.BuildResult> = new Map();
+
+      build.onResolve(
+        { filter: /worker-loader:.*\.worker.[jt]sx?$/ },
+        (args) => {
+          const importPath = args.path.split('worker-loader:')[1];
+          const importAbsolutePath = path.join(args.resolveDir, importPath);
+
+          // Pin the file extension to .js
+          const workerAbsolutePath = path.join(
+            path.dirname(importAbsolutePath),
+            path.basename(importAbsolutePath).replace(/\.[jt]sx?$/, '.js'),
+          );
+
+          // Path passed to esbuild must be either absolute or fully relative.
+          // If it's absolute, snapshots will fail because the built file will contain an absolute (different) path
+          // We need to make it relative to the current working directory, and add './'
+          // because esbuild doesn't like paths like 'packages/appName/src/...'
+          const relativePath =
+            './' + path.relative(process.cwd(), workerAbsolutePath);
```

Comment:
What is process.cwd() going to be? Does this mean modular root? If so use `getModularRoot` to determine this.

Reply:
@LukeSheard  Solved by recreating the abs path from root + relative path and using it as esbuild entrypoint.